### PR TITLE
Fix check for Serial run

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Tests that check for running in serial need to get adapted to use info from the current spec, since there's no more separate running, thus the `ParallelTotal` will stay the same, even if the test is run in Serial mode

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve
I thought we already have some helper functions for this.

File: tests/infra_test.go
Comment (dhiller) [inline tests/infra_test.go:1441]: Since we are running all tests in this container as `Serial`, shouldn't we move this assertion into the `BeforeEach`?

Comment (fossedihelm) [reply tests/infra_test.go:1441]: That function is used on in the event expectation. To check that the event we are expecting does not come from a parallel execution, we want to be sure we are not receiving spurious events.
https://github.com/kubevirt/kubevirt/pull/8848 This PR should help in this way.

Comment (dhiller) [reply tests/infra_test.go:1441]: Ah, thanks for clarification about that. I was just thinking that since the container already has the `Serial` label, we could check it first in the BeforeEach. Otherwise I'd fear that this assertion would only be called for some tests. But I don't want to block on that.

**Review by dhiller — APPROVED** [review]

File: tests/infra_test.go
Comment (fossedihelm) [reply tests/infra_test.go:1441]: I would wait until the other PR is merged, and delete it directly :)

**Comment by xpivarc** [comment]
/hold 
Helping Tide.
The storage lane is currently broken. https://github.com/kubevirt/kubevirt/pull/9135 should fix this.
Feel free to remove the hold once this is merged.

**Pull request merged**
